Mathlib/Data/List/Commutativity.lean

```diff
@@ -0,0 +1,104 @@
+/-
+Copyright (c) 2025 Shimanonogov Igor. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Shimanonogov Igor
+-/
+import Mathlib.Data.List.Defs
+import Mathlib.Data.List.RepeatSelf
+
+/-!
+# commutativity
+
+Results commencted with commutativity of List with respect to append.
+
+## Main results
+
+- `comm_iff_common_root`: Lyndon-Schutzenberger theorem.
+-/
+
+namespace List
+
+variable {α : Type*} (n m : ℕ) {u v : List α}
+
+theorem prefix_if_concat_comm : (u ++ v = v ++ u) → (v <+: u ∨ u <+: v) := by
+  intro h
+  rw [List.append_eq_append_iff] at h
+  cases h with
+  | inl r =>
+    right
+    obtain ⟨as, h₁, h₂⟩ := r
+    rw [h₁]
+    exact List.prefix_append u as
+  | inr l =>
+    left
+    obtain ⟨as, h₁, h₂⟩ := l
+    rw [h₁]
+    exact List.prefix_append v as
+
+theorem comm_iff_common_root {q p : List α} :
```

Comment:
Is this the L-S theorem: then add a doc-string please, mentioning this name. Thanks!